Automattic/woocommerce-payments
# Fix KYC treatment mode API and menu issues

Fixes #4171

#### Changes proposed in this Pull Request
- Avoid redirecting the settings menu to onboarding page.
- Redirect WC Admin task to onboarding page rather than settings, that then will redirect to connect page.
- Use user token for new onboarding requests. I'm still concerned about this one, not sure why it was working fine before. Although looking at the server controller looks like we should be using it from the beginning.

#### Testing instructions
- Get treatment bookmarklet from `woo_wcpayments_tasklist_click_introducing_select_business_type_202203_v1` experiment in Abacus.
- You'll need a non onboarded local site with **WCPay Dev > Proxy WPCOM requests** and **WooCommerce > Settings > Advanced > Allow usage of WooCommerce to be tracked** enabled.
- Run the treatment bookmarklet, it should show a success message.
- Go to **WooCommerce > Set up payments** task, click on **Finish Setup** and then on **Set up**.
- Should be redirected to new onboarding page with the KYC form.
- The form should load countries and filling it should render the required verification details.
- Complete the onboard.
- Check that the settings menu **Payments > Settings** points to `?page=wc-settings&tab=checkout&section=woocommerce_payments` and not `?page=wc-admin&path=/payments/onboarding`

-------------------


**Post merge**
QA Testing Not Applicable
